### ISSUE TITLE
Enter key on hardware android/iOS keyboard does not post message

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#8797]
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-10601)

GitHub issue: https://github.com/mattermost/mattermost-mobile/issues/784

Changes should be made to https://github.com/mattermost/mattermost-mobile repo

Summary
Enter key on hardware android keyboards does not post messages. Right now enter key changes line inside the message which is not useful at all while in conversations.

Environment Information
Device Name: Google Pixel C + Keyboard
OS Version: Android 7.1.2
Mattermost App Version: Latest Mattermost Android client as well Latest Mattermost Beta Android
Mattermost Server Version: 4.0
Steps to reproduce
In Android, try to post a message using enter hardware keyboard key. You'll need a hardware keyboard to replicate.

Expected behavior
Enter should post the message and Shift-Enter should change line inside the message (like on desktop clients)

Observed behavior
Android hardware keyboard Enter key changes line inside a message. Shift-Enter does the same.

**Comment by oljw** [comment]
Hello,
I'd like to try this issue if I could!
Thank you!

**Comment by oljw** [comment]
Hello,

While investigating the issue, I wanted to have some clarifications:

- As far as understand, the application navigation is supported by NavigationActivity in 'react-native-navigation' module that'd been modified and forked by '@migbot' from the original. 
- However, the MainActivity is extended from a SplashActivity which is destroyed after the splash screen is finished.
- In this case, I couldn't play around with 'onConfigurationChanged', 'onKeyUp', or 'onKeyDown' from the activity.
- As I've been searching the Internet, there is no way to detect if an external keyboard is connected nor which specific key had been entered by just using js. So it is necessary to communicate with native modules.
- In order to do that, I need to communicate with an Activity's 'onKeyUp' or 'onKeyDown' method.

Q: So I was looking for a way to do this without modifying the structure of the application and thinking you guys might already have something to communicate these kind of functions between js and native part. Do you? If so, which class or module should I use?

**Comment by hanzei** [comment]
@migbot Could you please take a look at the above question?

**Comment by migbot** [comment]
@oljw There's actually an easier way to do this with [blurOnSubmit](https://facebook.github.io/react-native/docs/textinput#bluronsubmit), however, let me ask @enahum if there is any reason why `blurOnSubmit` is set to false [here](https://github.com/mattermost/mattermost-mobile/commit/90dc6ecb254a451094dda47fe87dc637b2b5042f#diff-dd0ced5790e5b9b6ca483a2770520a43R398).

**Comment by migbot** [comment]
@oljw @enahum Disregard my comment above. For one, while setting `blurOnSubmit` to true allows the enter key to trigger `onSubmitEditing`, a `shift-enter` would need to be caught in `onKeyPress`. However, [onKeyPress](https://facebook.github.io/react-native/docs/textinput#onkeypress) only handles inputs from a soft keyboard on Android.

@oljw Let me digest your initial comment and questions a bit longer. I will reply shortly.

**Comment by oljw** [comment]
@migbot Alright, cool! Thanks for replying.

**Comment by migbot** [comment]
@oljw OK, take a look at this [comment](https://github.com/wix/react-native-navigation/issues/3258#issuecomment-392510061) describing how to respond to key events in react-native-navigation v1. Now it looks like the only function you can override is [onKeyUp](https://github.com/migbot/react-native-navigation/blob/03c623c373f818827a463ca0fe90f86f56e71b2f/android/app/src/main/java/com/reactnativenavigation/controllers/ActivityCallbacks.java). If you need to override other functions, let me get @enahum's thoughts regarding making changes to our fork, since we are planning to move away from that version of the navigation library.

**Comment by oljw** [comment]
@migbot Cool cool I think I can try to work with that. I'll let you know if anything else come up.
Thanks for your help!

**Comment by oljw** [comment]
Hi all,

I've submitted a PR for the Android side of this issue:
https://github.com/mattermost/mattermost-mobile/pull/2676

Now, I'm trying to implement this functionality to the iOS side at the moment.
I will have to work with the native iOS side similar how I implemented the Android side because the 'onKeyPressed' function at js side cannot detect the modifier keys such as 'shift'.
While at it, I'm trying to get a callback from the 'UIKeyCommand' when 'shift' key is pressed.
However, in order to get a callback from 'UIKeyCommand', there must be a view that is extended from UIView or a view controller so I can override the method. As I see the iOS code, I'm not able to see any of view controller class except for the RCTRootView which I cannot directly control.

I am kind of stuck at the moment and would like to get any tips from you professionals like @migbot .
